src/agent/thread_helper.cpp

```diff
@@ -0,0 +1,323 @@
+#include "agent/thread_helper.hpp"
+
+#include <assert.h>
+#include <string.h>
+
+#include <string>
+
+#include "openthread/border_router.h"
+#include "openthread/channel_manager.h"
+#include "openthread/jam_detection.h"
+#include "openthread/joiner.h"
+#include "openthread/platform/radio.h"
+#include "openthread/thread_ftd.h"
+
+#include "agent/ncp_openthread.hpp"
+#include "common/code_utils.hpp"
+
+namespace otbr {
+namespace agent {
+
+ThreadHelper::ThreadHelper(otInstance *aInstance, ot::BorderRouter::Ncp::ControllerOpenThread *aNcp)
+    : mInstance(aInstance)
+    , mNcp(aNcp)
+    , mScanHandler(nullptr)
+{
+}
+
+otError ThreadHelper::Init(void)
+{
+    srand(static_cast<unsigned int>(time(nullptr)));
+
+    return otSetStateChangedCallback(mInstance, sStateChangedCallback, this);
+}
+
+void ThreadHelper::sStateChangedCallback(otChangedFlags aFlags, void *aContext)
```

Comment:
Why don't use `ThreadHelper*` as the parameter type for `aContext` here, if you are casting it immediately?

Reply:
The sStateChangedCallback is passed to openthread so I cannot modify its function signature.